b2sdk/api.py

```diff
@@ -462,19 +478,32 @@ def get_file_info(self, file_id: str) -> Dict[str, Any]:
         """
         return self.session.get_file_info_by_id(file_id)
 
-    def check_bucket_restrictions(self, bucket_name):
+    def check_bucket_name_restrictions(self, bucket_name: str):
         """
         Check to see if the allowed field from authorize-account has a bucket restriction.
 
         If it does, checks if the bucket_name for a given api call matches that.
         If not, it raises a :py:exc:`b2sdk.v1.exception.RestrictedBucket` error.
 
-        :param str bucket_name: a bucket name
         :raises b2sdk.v1.exception.RestrictedBucket: if the account is not allowed to use this bucket
         """
+        self._check_bucket_restrictions('bucketName', bucket_name)
+
+    def check_bucket_id_restrictions(self, bucket_id: str):
+        """
+        Check to see if the allowed field from authorize-account has a bucket restriction.
+
+        If it does, checks if the bucket_id for a given api call matches that.
+        If not, it raises a :py:exc:`b2sdk.v1.exception.RestrictedBucket` error.
+
+        :raises b2sdk.v1.exception.RestrictedBucket: if the account is not allowed to use this bucket
+        """
+        self._check_bucket_restrictions('bucketId', bucket_id)
+
+    def _check_bucket_restrictions(self, key, value):
         allowed = self.account_info.get_allowed()
-        allowed_bucket_name = allowed['bucketName']
+        allowed_bucket_identifier = allowed[key]
```

Comment:
this kind of sounds like it could explode if I have a bucket with `name=bucket1, id=abcde` and another bucket with `name=abcde, id=bcdef`